File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/responsedata/PieceAddedEventResponseData.java
package com.kafkaquest.kq.common.util.models.events.responsedata;

import com.kafkaquest.kq.common.util.models.events.other.TransportScoreCard;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PieceAddedEventResponseData extends ResponseData {
    private UUID pieceId;
    private int movedRowPosition;
    private int movedColumnPosition;
    private boolean hasGameEnded;
    private TransportScoreCard scoreCard;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Events/PieceAddedEventResponseData.cs
using KQ_Projection.Common.Models.Jigsaw;
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events
{
    public class PieceAddedEventResponseData : ResponseData
    {
        [JsonPropertyName("pieceId")]
        public Guid pieceId { get; set; }

        [JsonPropertyName("movedRowPosition")]
        public int movedRowPosition { get; set; }

        [JsonPropertyName("movedColumnPosition")]
        public int movedColumnPosition { get; set; }    

        [JsonPropertyName("hasGameEnded")]
        public bool hasGameEnded { get; set; }

        [JsonPropertyName("scoreCard")]
        public TransportScoreCard scoreCard { get; set; }
    }
}
